src/test/java/io/xstefank/wildfly/bot/PRDirectoriesVerificationTest.java

```diff
@@ -0,0 +1,180 @@
+                })
+                .when().payloadFromString(gitHubJson.jsonString())
+                .event(GHEvent.PULL_REQUEST)
+                .then().github(mocks -> {
+                    GHRepository repo = mocks.repository(TestConstants.TEST_REPO);
+                    Mockito.verify(repo).getDirectoryContent("src");
+                    Mockito.verify(repo).createCommitStatus(gitHubJson.commitSHA(),
+                            GHCommitState.SUCCESS, "", "Valid", "Configuration File");
+                });
+    }
+
+    @Test
+    public void nonExistingDirectoryTest() throws IOException {
+        mockedContext = MockedGHPullRequestProcessor.builder(gitHubJson.id())
+                .prFile(".github/wildfly-bot.yml");
+        given().github(mocks -> {
+                    Util.mockContext(mocks, mockedContext);
+                    GHRepository repo = mocks.repository(TestConstants.TEST_REPO);
+                    GHContent mockGHContent = mock(GHContent.class);
+                    when(repo.getFileContent(".github/" + RuntimeConstants.CONFIG_FILE_NAME, gitHubJson.commitSHA())).thenReturn(mockGHContent);
+                    when(mockGHContent.read()).thenReturn(IOUtils.toInputStream("""
+                                    wildfly:
+                                      rules:
+                                        - id: "id"
+                                          directories: [src]""",
+                            "UTF-8"));
+
+                })
+                .when().payloadFromString(gitHubJson.jsonString())
+                .event(GHEvent.PULL_REQUEST)
+                .then().github(mocks -> {
+                    GHRepository repo = mocks.repository(TestConstants.TEST_REPO);
+                    Mockito.verify(repo).getDirectoryContent("src");
+                    Mockito.verify(repo).createCommitStatus(gitHubJson.commitSHA(),
+                            GHCommitState.ERROR, "", "Rule is missing an id or multiple rules have the same id.", "Configuration File");
+                });
+    }
+
+    @Test
+    public void existingFileTest() throws IOException {
+        mockedContext = MockedGHPullRequestProcessor.builder(gitHubJson.id())
+                .prFile(".github/wildfly-bot.yml")
+                .repoFiles(Set.of("pom.xml"));
+        given().github(mocks -> {
+                    Util.mockContext(mocks, mockedContext);
+                    GHRepository repo = mocks.repository(TestConstants.TEST_REPO);
+                    GHContent mockGHContent = mock(GHContent.class);
+                    when(repo.getFileContent(".github/" + RuntimeConstants.CONFIG_FILE_NAME, gitHubJson.commitSHA())).thenReturn(mockGHContent);
+                    when(mockGHContent.read()).thenReturn(IOUtils.toInputStream("""
+                                    wildfly:
+                                      rules:
+                                        - id: "id"
+                                          directories: [pom.xml]""",
```

Comment:
this also feels wrong

Reply:
Should we disallow usage of files all together? However that might not be that easy, as some files might be missing extensions or directories might be using `.directory`.
I wanted to test this just in case, as it throws an exception, which is caught and so I didn't want a  file to cause an error in the rule.